### PR TITLE
LPS-118086 Extract providers to own modules

### PR DESCRIPTION
**Pull request by csierra**
Hey Brian, /cc @arthurchan35

here we are extracting the different providers to their own modules and making the `CryptoHashGeneratorImpl` accept the provider as a dependency, so different algorithms can be supported. 

Let us know if the structure is ok so far. 

Carlos.

### CODE REVIEW

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/asset_list/source.jsp [hunk 1]
```diff
@@ -234,427 +235,92 @@ List<AssetRendererFactory<?>> classTypesAssetRendererFactories = new ArrayList<>
...
-		var ddmStructureFieldMessageContainer = document.getElementById(
-			'<portlet:namespace />' + className + 'ddmStructureFieldMessage'
-		);
-
-		if (ddmStructureFieldMessageContainer) {
-			ddmStructureFieldMessageContainer.innerHTML = Liferay.Util.escape(
-				message
-			);
-		}
-	}
```
Comment (liferay-appsec) [inline modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/asset_list/source.jsp:647]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/asset_list/source.jsp [hunk 2]
```diff
@@ -234,427 +235,92 @@ List<AssetRendererFactory<?>> classTypesAssetRendererFactories = new ArrayList<>
...
+		<div class="asset-subtypefield-selected <%= Validator.isNull(editAssetListDisplayContext.getDDMStructureFieldName()) ? "hide" : StringPool.BLANK %>">
+			<aui:input name="TypeSettingsProperties--ddmStructureFieldName--" type="hidden" value="<%= editAssetListDisplayContext.getDDMStructureFieldName() %>" />
 
-	Util.toggleSelectBox(
-		'<portlet:namespace />anyAssetType',
-		'false',
-		'<portlet:namespace />classNamesBoxes'
```
Comment (liferay-appsec) [inline modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/asset_list/source.jsp:652]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/asset_list/source.jsp [hunk 3]
```diff
@@ -234,427 +235,92 @@ List<AssetRendererFactory<?>> classTypesAssetRendererFactories = new ArrayList<>
...
-	);
+			<aui:input name="TypeSettingsProperties--ddmStructureFieldValue--" type="hidden" value="<%= editAssetListDisplayContext.getDDMStructureFieldValue() %>" />
 
-	function removeListener() {
```
Comment (liferay-appsec) [inline modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/asset_list/source.jsp:655]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/commerce/commerce-health-status-web/src/main/java/com/liferay/commerce/health/status/web/internal/ServiceAccessPolicyCommerceHealthHttpStatus.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/commerce/commerce-health-status-web/src/main/java/com/liferay/commerce/health/status/web/internal/ServiceAccessPolicyCommerceHealthHttpStatus.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/commerce/commerce-lang/src/main/resources/content/Language.properties [hunk 5]
```diff
@@ -1284,6 +1284,8 @@ send-to=Send To
 sent-to=Sent To
```
Comment (liferay-appsec) [inline modules/apps/commerce/commerce-lang/src/main/resources/content/Language.properties:1284]: Check that added properties are either titles or sentences

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/edit_form_instance.jsp [hunk 6]
```diff
@@ -144,11 +144,11 @@ renderResponse.setTitle((formInstance == null) ? LanguageUtil.get(request, "new-
 		<div class="ddm-form-basic-info">
 			<clay:container-fluid>
 				<h1>
-					<aui:input autoSize="<%= true %>" cssClass="ddm-form-name ddm-placeholder hidden" label="" name="nameEditor" placeholder='<%= LanguageUtil.get(request, "untitled-form") %>' type="textarea" value="<%= HtmlUtil.escapeAttribute(ddmFormAdminDisplayContext.getFormName()) %>" />
+					<aui:input autoSize="<%= true %>" cssClass="ddm-form-name ddm-placeholder hidden" label="" name="nameEditor" placeholder='<%= LanguageUtil.get(request, "untitled-form") %>' type="textarea" value="<%= HtmlUtil.toInputSafe(ddmFormAdminDisplayContext.getFormName()) %>" />
```
Comment (liferay-appsec) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/edit_form_instance.jsp:147]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/edit_form_instance.jsp [hunk 7]
```diff
@@ -144,11 +144,11 @@ renderResponse.setTitle((formInstance == null) ? LanguageUtil.get(request, "new-
...
 				</h1>
 
 				<h5>
-					<aui:input autoSize="<%= true %>" cssClass="ddm-form-description ddm-placeholder hidden" label="" name="descriptionEditor" placeholder='<%= LanguageUtil.get(request, "add-a-short-description-for-this-form") %>' type="textarea" value="<%= HtmlUtil.escapeAttribute(ddmFormAdminDisplayContext.getFormDescription()) %>" />
+					<aui:input autoSize="<%= true %>" cssClass="ddm-form-description ddm-placeholder hidden" label="" name="descriptionEditor" placeholder='<%= LanguageUtil.get(request, "add-a-short-description-for-this-form") %>' type="textarea" value="<%= HtmlUtil.toInputSafe(ddmFormAdminDisplayContext.getFormDescription()) %>" />
```
Comment (liferay-appsec) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/edit_form_instance.jsp:151]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-provider/portal-crypto-hash-provider-bcrypt/src/main/java/com/liferay/portal/crypto/hash/provider/bcrypt/internal/BCryptCryptoHashProvider.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-provider/portal-crypto-hash-provider-bcrypt/src/main/java/com/liferay/portal/crypto/hash/provider/bcrypt/internal/BCryptCryptoHashProvider.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-provider/portal-crypto-hash-provider-message-digest/src/main/java/com/liferay/portal/crypto/hash/provider/message/digest/internal/MessageDigestCryptoHashProvider.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-provider/portal-crypto-hash-provider-message-digest/src/main/java/com/liferay/portal/crypto/hash/provider/message/digest/internal/MessageDigestCryptoHashProvider.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/facet/NestedFacetProcessor.java [hunk 10]
```diff
@@ -0,0 +1,100 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/facet/NestedFacetProcessor.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/NestedFacetImpl.java [hunk 11]
```diff
@@ -0,0 +1,107 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/NestedFacetImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/nested/NestedFacetFactoryImpl.java [hunk 12]
```diff
@@ -0,0 +1,43 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/nested/NestedFacetFactoryImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/nested/NestedFacetSearchContributorImpl.java [hunk 13]
```diff
@@ -0,0 +1,178 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/facet/nested/NestedFacetSearchContributorImpl.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/resources/content/Language.properties [hunk 14]
```diff
@@ -20,6 +20,8 @@ provider-name-help=Set the name for the OpenID Connect provider.
 scopes-help=Set the scopes Liferay will request during authentication. Scopes are delimited with spaces.
```
Comment (liferay-appsec) [inline modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/resources/content/Language.properties:20]: Check that added properties are either titles or sentences

File: modules/dxp/apps/analytics/analytics-reports-journal-impl/src/main/java/com/liferay/analytics/reports/journal/internal/info/item/provider/JournalArticleAnalyticsReportsInfoItemObjectProvider.java [hunk 15]
```diff
@@ -0,0 +1,53 @@
+/**
```
Comment (liferay-appsec) [inline modules/dxp/apps/analytics/analytics-reports-journal-impl/src/main/java/com/liferay/analytics/reports/journal/internal/info/item/provider/JournalArticleAnalyticsReportsInfoItemObjectProvider.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by csierra** [comment]
ci:forward

**Comment by csierra** [comment]
ci:test:sf

**Comment by csierra** [comment]
ci:stop

**Comment by csierra** [comment]
ci:forward

**Comment by arthurchan35** [comment]
Fixed unresolved requirement and update ticket number:
https://github.com/liferay-appsec/liferay-portal/pull/272

**Comment by csierra** [comment]
thx a ton @arthurchan35!

**Pull request closed without merging**